utilities/transactions/write_unprepared_txn_db.h

```diff
@@ -11,16 +11,27 @@
 #endif
 
 #include "utilities/transactions/write_prepared_txn_db.h"
-
 #include "utilities/transactions/write_unprepared_txn.h"
 
 namespace rocksdb {
 
+class WriteUnpreparedTxn;
+
 class WriteUnpreparedTxnDB : public WritePreparedTxnDB {
+ public:
   using WritePreparedTxnDB::WritePreparedTxnDB;
 
   Transaction* BeginTransaction(const WriteOptions& write_options, const TransactionOptions& txn_options,
                                 Transaction* old_txn) override;
+
+  // Struct to hold ownership of snapshot and read callback for cleanup.
+  struct IteratorState;
+
+  using WritePreparedTxnDB::NewIterator;
```

Comment:
How about NewIterators? Could be another PR to make the review easier.

Reply:
I don't think NewIterators needs an override because it's only available on the DB, not transaction. On the DB, we don't need to worry about read-your-own-write.